File: support-bot.js
const { Telegraf } = require('telegraf');

const BOT_TOKEN = process.env.BOT_TOKEN;
const ADMIN_ID = parseInt(process.env.ADMIN_ID);

if (!BOT_TOKEN || !ADMIN_ID) {
    console.error('❌ Ошибка: BOT_TOKEN или ADMIN_ID не заданы');
    process.exit(1);
}

const bot = new Telegraf(BOT_TOKEN);

// Хранилище: какой пользователь ожидает ответа от админа
let waitingForReply = {};

bot.start((ctx) => {
    ctx.reply('👋 Добро пожаловать в службу поддержки!\n\nОпишите вашу проблему или отправьте скриншот — я передам администратору.');
});

bot.help((ctx) => {
    ctx.reply('📌 Просто напишите сообщение или отправьте фото — администратор получит и сможет ответить.');
});

// Обработка фото (скриншотов)
bot.on('photo', async (ctx) => {
    const userId = ctx.from.id;
    const username = ctx.from.username || `id${userId}`;
    const caption = ctx.message.caption || 'Без текста';
    
    // Берём самое большое фото (последнее в массиве)
    const photo = ctx.message.photo[ctx.message.photo.length - 1];
    const fileId = photo.file_id;

    try {
        await bot.telegram.sendPhoto(
            ADMIN_ID,
            fileId,
            {
                caption: `📸 Скриншот\n👤 @${username}\n🆔 ${userId}\n💬 ${caption}`,
                reply_markup: {
                    inline_keyboard: [
                        [{ text: '✏️ Ответить', callback_data: `reply_${userId}` }],
                        [{ text: '🚫 Игнорировать', callback_data: `ignore_${userId}` }]
                    ]
                }
            }
        );
        await ctx.reply('✅ Скриншот отправлен администратору.');
    } catch (err) {
        console.error('Ошибка отправки фото:', err.message);
        await ctx.reply('❌ Ошибка отправки скриншота.');
    }
});

// Обработка документов (файлов)
bot.on('document', async (ctx) => {
    const userId = ctx.from.id;
    const username = ctx.from.username || `id${userId}`;
    const caption = ctx.message.caption || 'Без текста';
    const fileId = ctx.message.document.file_id;

    try {
        await bot.telegram.sendDocument(
            ADMIN_ID,
            fileId,
            {
                caption: `📎 Файл\n👤 @${username}\n🆔 ${userId}\n💬 ${caption}`,
                reply_markup: {
                    inline_keyboard: [
                        [{ text: '✏️ Ответить', callback_data: `reply_${userId}` }],
                        [{ text: '🚫 Игнорировать', callback_data: `ignore_${userId}` }]
                    ]
                }
            }
        );
        await ctx.reply('✅ Файл отправлен администратору.');
    } catch (err) {
        console.error('Ошибка отправки файла:', err.message);
        await ctx.reply('❌ Ошибка отправки файла.');
    }
});

// Обработка текстовых сообщений
bot.on('text', async (ctx) => {
    const userId = ctx.from.id;
    const messageText = ctx.message.text;

    // Если сообщение от администратора
    if (userId === ADMIN_ID) {
        if (waitingForReply[userId]) {
            const targetUserId = waitingForReply[userId];
            delete waitingForReply[userId];

            try {
                await bot.telegram.sendMessage(
                    targetUserId,
                    `📬 *Ответ от поддержки:*\n\n${messageText}`,
                    { parse_mode: 'Markdown' }
                );
                await ctx.reply(`✅ Ответ отправлен пользователю ${targetUserId}.`);
            } catch (err) {
                await ctx.reply(`❌ Ошибка отправки: ${err.message}`);
                waitingForReply[userId] = targetUserId;
            }
            return;
        }
        return;
    }

    // Если сообщение от обычного пользователя
    const username = ctx.from.username || `id${userId}`;
    
    try {
        await bot.telegram.sendMessage(
            ADMIN_ID,
            `🆕 Новое обращение\n👤 @${username}\n🆔 ${userId}\n\n💬 ${messageText}`,
            {
                reply_markup: {
                    inline_keyboard: [
                        [{ text: '✏️ Ответить', callback_data: `reply_${userId}` }],
                        [{ text: '🚫 Игнорировать', callback_data: `ignore_${userId}` }]
                    ]
                }
            }
        );
        await ctx.reply('✅ Сообщение отправлено администратору.');
    } catch (err) {
        console.error('Ошибка:', err.message);
        await ctx.reply('❌ Ошибка отправки.');
    }
});

// Админ нажимает «Ответить»
bot.action(/reply_(.+)/, async (ctx) => {
    if (ctx.from.id !== ADMIN_ID) {
        await ctx.answerCbQuery('Нет доступа');
        return;
    }

    const targetUserId = parseInt(ctx.match[1]);
    waitingForReply[ADMIN_ID] = targetUserId;
    
    await ctx.editMessageText(`✏️ Введите ответ для пользователя ${targetUserId}:\n\nПросто напишите сообщение в этот чат.`);
    await ctx.answerCbQuery();
});

// Админ нажимает «Игнорировать»
bot.action(/ignore_(.+)/, async (ctx) => {
    if (ctx.from.id !== ADMIN_ID) {
        await ctx.answerCbQuery('Нет доступа');
        return;
    }

    const targetUserId = parseInt(ctx.match[1]);
    
    try {
        await bot.telegram.sendMessage(
            targetUserId,
            '⏳ Ваше обращение находится в обработке. Пожалуйста, ожидайте.'
        );
    } catch (err) {
        console.error('Ошибка отправки уведомления:', err.message);
    }
    
    await ctx.editMessageCaption(`🚫 Обращение от пользователя ${targetUserId} помечено как обработанное.`);
    await ctx.answerCbQuery('Уведомление отправлено пользователю');
});

const express = require('express');
const app = express();
const port = process.env.PORT || 10000;
app.get('/', (req, res) => res.send('Bot is running'));
app.listen(port, () => console.log(`HTTP server on port ${port}`));

bot.launch().then(() => console.log('🚀 Бот успешно запущен')).catch(err => {
    console.error('❌ Ошибка запуска:', err);
    process.exit(1);
});

File: support_bot.py
import asyncio
import logging
from aiogram import Bot, Dispatcher, types
from aiogram.filters import Command
from aiogram.types import InlineKeyboardMarkup, InlineKeyboardButton
from aiogram.fsm.context import FSMContext
from aiogram.fsm.state import State, StatesGroup
import os

# 🔐 Токен твоего бота (получи у @BotFather)
BOT_TOKEN = "8714370698:AAEtod0QulFufvb1DE_h_TwRsPsVV_dfnt4"
# 🧑‍💻 ID админа (твой Telegram ID)
ADMIN_ID = 1444520038  # твой ID из базы

bot = Bot(token=BOT_TOKEN)
dp = Dispatcher()

# Хранилище временных сообщений от пользователей
temp_messages = {}

# Состояния для ответа админа
class AdminReply(StatesGroup):
    waiting_for_reply = State()

# ---------- Команда /start ----------
@dp.message(Command("start"))
async def cmd_start(message: types.Message):
    await message.answer(
        "👋 Добро пожаловать в службу поддержки!\n\n"
        "Опишите вашу проблему — я передам её администратору.\n"
        "Обычно ответ приходит в течение 24 часов."
    )

# ---------- Команда /help ----------
@dp.message(Command("help"))
async def cmd_help(message: types.Message):
    await message.answer(
        "📌 *Как связаться с поддержкой:*\n"
        "Просто напишите мне любое сообщение — я передам его администратору.\n\n"
        "После ответа администратора вы получите уведомление.",
        parse_mode="Markdown"
    )

# ---------- Обработка любого текста от НЕ админа ----------
@dp.message(lambda msg: msg.from_user.id != ADMIN_ID)
async def forward_to_admin(message: types.Message):
    user = message.from_user
    user_name = user.username or f"{user.first_name} {user.last_name or ''}".strip() or "Без имени"
    user_link = f"@{user.username}" if user.username else f"[{user_name}](tg://user?id={user.id})"
    
    # Отправляем админу
    keyboard = InlineKeyboardMarkup(inline_keyboard=[
        [InlineKeyboardButton(text="✏️ Ответить", callback_data=f"reply_{user.id}")]
    ])
    
    await bot.send_message(
        ADMIN_ID,
        f"🆕 *Новое обращение*\n"
        f"👤 Пользователь: {user_link}\n"
        f"🆔 ID: `{user.id}`\n\n"
        f"💬 *Текст:*\n{message.text}",
        reply_markup=keyboard,
        parse_mode="Markdown"
    )
    
    # Подтверждение пользователю
    await message.answer("✅ Ваше сообщение отправлено администратору. Ожидайте ответа в ближайшее время.")

# ---------- Обработка нажатия кнопки "Ответить" админом ----------
@dp.callback_query(lambda c: c.data.startswith("reply_"))
async def ask_reply(callback: types.CallbackQuery, state: FSMContext):
    user_id = int(callback.data.split("_")[1])
    await state.update_data(user_id=user_id)
    await state.set_state(AdminReply.waiting_for_reply)
    
    await callback.message.answer(
        f"✍️ Введите ответ для пользователя (ID: {user_id}):"
    )
    await callback.answer()

# ---------- Получение ответа от админа ----------
@dp.message(AdminReply.waiting_for_reply)
async def send_reply(message: types.Message, state: FSMContext):
    data = await state.get_data()
    user_id = data.get("user_id")
    reply_text = message.text
    
    if not user_id:
        await message.answer("❌ Ошибка: не удалось определить пользователя.")
        await state.clear()
        return
    
    # Отправляем ответ пользователю
    try:
        await bot.send_message(
            user_id,
            f"📬 *Ответ от поддержки:*\n\n{reply_text}\n\n"
            "Если остались вопросы — напишите новое сообщение.",
            parse_mode="Markdown"
        )
        await message.answer(f"✅ Ответ отправлен пользователю {user_id}.")
    except Exception as e:
        await message.answer(f"❌ Ошибка отправки: {e}")
    
    await state.clear()

# ---------- Админ может написать пользователю вручную (команда /reply 123456 текст) ----------
@dp.message(Command("reply"))
async def admin_direct_reply(message: types.Message):
    if message.from_user.id != ADMIN_ID:
        await message.answer("⛔ Доступно только администратору.")
        return
    
    parts = message.text.split(maxsplit=2)
    if len(parts) < 3:
        await message.answer("❌ Использование: /reply USER_ID текст")
        return
    
    _, user_id_str, reply_text = parts
    user_id = int(user_id_str)
    
    try:
        await bot.send_message(
            user_id,
            f"📬 *Ответ от поддержки:*\n\n{reply_text}\n\n"
            "Если остались вопросы — напишите новое сообщение.",
            parse_mode="Markdown"
        )
        await message.answer(f"✅ Сообщение отправлено пользователю {user_id}.")
    except Exception as e:
        await message.answer(f"❌ Ошибка: {e}")

# ---------- Запуск ----------
async def main():
    print("🚀 Бот запущен...")
    await dp.start_polling(bot)

if __name__ == "__main__":
    asyncio.run(main())
